py/builtinimport.c

```diff
@@ -190,40 +190,46 @@ STATIC void do_load(mp_obj_t module_obj, vstr_t *file) {
     char *file_str = vstr_null_terminated_str(file);
     #endif
 
-    // If we support frozen modules (either as str or mpy) then try to find the
-    // requested filename in the list of frozen module filenames.
-    #if MICROPY_MODULE_FROZEN
-    void *modref;
-    int frozen_type = mp_find_frozen_module(file_str, file->len, &modref);
-    #endif
+    #if MICROPY_MODULE_FROZEN || MICROPY_PERSISTENT_CODE_LOAD
+    if (strncmp(MP_FROZEN_FAKE_DIR_SLASH,
+                file_str,
+                MP_FROZEN_FAKE_DIR_SLASH_LENGTH) == 0) {
+        // If we support frozen modules (either as str or mpy) then try to find the
+        // requested filename in the list of frozen module filenames.
+        #if MICROPY_MODULE_FROZEN
+        void *modref;
+        int frozen_type = mp_find_frozen_module(file_str + MP_FROZEN_FAKE_DIR_SLASH_LENGTH, file->len - MP_FROZEN_FAKE_DIR_SLASH_LENGTH, &modref);
```

Comment:
I think the checking for ".frozen/" here now means the checking for ".frozen/" in `mp_find_frozen_module()` is no longer needed. But we can just leave it in. Basically you hoisted that check up to where it was really needed.

This and `pyexec_frozen_module()` are the only call to `mp_find_frozen_module()`.